api/omnitransaction.py

```diff
@@ -0,0 +1,124 @@
+            rawtx = createrawtx_reference(self.rawdata['transaction_to'], rawtx)['result']
+
+        # Add the payload    
+        if len(payload) <= 160:  #80bytes
+            rawtx = createrawtx_opreturn(payload, rawtx)['result']
+        else:
+            rawtx = createrawtx_multisig(payload, rawtx)['result']
+
+        # Decode transaction to get total needed amount
+        decodedtx = decoderawtransaction(rawtx)['result']
+
+        # Sumup the outputs
+        fee_total = Decimal(self.fee)
+        for output in decodedtx['vout']:
+            fee_total += Decimal(output['value'])
+
+        fee_total_satoshi = int( round( fee_total * Decimal(1e8) ) )
+
+        # Get utxo to generate inputs
+        print "Calling bc_getutxo with ", self.rawdata['transaction_from'], fee_total_satoshi
+        dirty_txes = bc_getutxo( self.rawdata['transaction_from'], fee_total_satoshi )
+        print "received", dirty_txes
+
+        if (dirty_txes['error'][:3]=='Con'):
+            raise Exception({ "status": "NOT OK", "error": "Couldn't get list of unspent tx's. Response Code: " + dirty_txes['code']  })
+
+        if (dirty_txes['error'][:3]=='Low'):
+            raise Exception({ "status": "NOT OK", "error": "Not enough funds, try again. Needed: " + str(fee_total) + " but Have: " + dirty_txes['avail']  })
+
+        total_amount = dirty_txes['avail']
+        unspent_tx = dirty_txes['utxos']
+
+        change = total_amount - fee_total_satoshi
+
+        #DEBUG 
+        print [ "Debugging...", dirty_txes,"miner fee sats: ", self.fee, "change: ",change,"total_amt: ", total_amount,"fee tot sat: ", fee_total_satoshi,"utxo ",  unspent_tx,"to ", self.rawdata['transaction_to'] ]
+
+        #source script is needed to sign on the client credit grazcoin
+        hash160=bc_address_to_hash_160(self.rawdata['transaction_from']).encode('hex_codec')
+        prevout_script='OP_DUP OP_HASH160 ' + hash160 + ' OP_EQUALVERIFY OP_CHECKSIG'
+
+        validnextinputs = []   #get valid redeemable inputs
+        for unspent in unspent_tx:
+            #retrieve raw transaction to spend it
+            prev_tx = getrawtransaction(unspent[0])['result']
+
+            for output in prev_tx['vout']:
+                if 'reqSigs' in output['scriptPubKey'] and output['scriptPubKey']['reqSigs'] == 1 and output['scriptPubKey']['type'] != 'multisig':
+                    for address in output['scriptPubKey']['addresses']:
+                        if address == self.rawdata['transaction_from'] and int(output['n']) == int(unspent[1]):
+                            validnextinputs.append({ "txid": prev_tx['txid'], "vout": output['n'], "scriptPubKey" : output['scriptPubKey']['hex'], "value" : output['value']})
+                            break
+        # Add the inputs
+        for input in validnextinputs:
+            rawtx = createrawtx_input(input['txid'],input['vout'],rawtx)['result']
+
+        # Add the change if above dust
+        if change > 5757:
```

Comment:
The check `change > 5757` can safely be removed, because `omni_createrawtx_change` doesn't add change, if it's considered as dust (and the original transaction is simply returned in this case, without error).